src/lib/operators/abstract_operator.hpp

```diff
@@ -161,6 +165,9 @@ class AbstractOperator : public std::enable_shared_from_this<AbstractOperator>,
   // Is nullptr until the operator is executed
   std::shared_ptr<const Table> _output;
 
+  // LQP node with which this operator has been created. Can be uninitialized.
+  std::shared_ptr<const AbstractLQPNode> _lqp_node;
```

Comment:
Right now, it probably doesn't make any difference as we have nothing pointing from the LQP to the PQP, but I think I would feel better if this was a weak_ptr

Reply:
Let's discuss in person. I see what you mean, but I am not sure that it helps us here.

Reply:
I hope to have addressed all your brought up issues.